adapters/v1/syft.go

```diff
@@ -0,0 +1,95 @@
+package v1
+
+import (
+	"bytes"
+	"context"
+	"sync"
+
+	"github.com/anchore/stereoscope/pkg/image"
+	"github.com/anchore/syft/syft"
+	"github.com/anchore/syft/syft/formats/spdxjson"
+	"github.com/anchore/syft/syft/pkg/cataloger"
+	"github.com/anchore/syft/syft/sbom"
+	"github.com/anchore/syft/syft/source"
+	"github.com/kubescape/kubevuln/core/domain"
+	"github.com/kubescape/kubevuln/core/ports"
+	"go.opentelemetry.io/otel"
+)
+
+type SyftAdapter struct {
+	mu sync.Mutex
+}
+
+var _ ports.SBOMCreator = (*SyftAdapter)(nil)
+
+func NewSyftAdapter() *SyftAdapter {
+	return &SyftAdapter{}
+}
+
+func (s *SyftAdapter) CreateSBOM(ctx context.Context, imageID string, options domain.RegistryOptions) (domain.SBOM, error) {
+	ctx, span := otel.Tracer("").Start(ctx, "CreateSBOM")
+	defer span.End()
+	// ensure only one SBOM is created at a time
+	s.mu.Lock()
+	defer s.mu.Unlock()
```

Comment:
question: why do we have to create only one SBOM at a time?

If I understand correctly, since we lock the mutex for the entire adapter, we would only be able to create an SBOM for one image at a time.

Why should we not have the ability to create SBOM for `image1` and `image2` concurrently?

Reply:
This is a requirement from @dwertent and @rcohencyberarmor because the image is downloaded during scan, they want to avoid disk usage issues.
The mutex is used (for the moment) because it's simpler regarding error reporting to the Operator than a pure event-driven solution (although they might want to go there).

Reply:
If that’s the path you’ve chosen, I don’t mind.

However, judging by the function docstring, if our goal here is to limit possible disk usage, a better solution might be using a semaphore instead of a mutex, since it will allow us to adjust the concurrency level as necessary.

Reply:
good idea :)